### DISCUSSION TITLE
Fixture Factory & CakeDC/Users

### DISCUSSION THREAD

**Discussion by Petrucheqa** [vierge-noire/cakephp-fixture-factories#160] [Q&A] [answered]
Recently I've faced an issue trying to use plugins CakeDC/Users & Fixture Factories.

My project is running CakePHP 4.3 and the latest versions for both plugins.

I've found that if you are using a `CustomUser` class that extends CakeDC/Users class and that you try to generate a `CustomUserFactory` it will throw this error : 

```
App\Test\TestCase\Controller\BankingProposalControllerTest::testViewProjectProposals
TypeError: class_exists(): Argument #1 ($class) must be of type string, null given

/app/vendor/cakedc/users/src/Model/Entity/User.php:113
/app/vendor/cakedc/users/src/Model/Entity/User.php:100
/app/vendor/cakedc/users/src/Model/Entity/User.php:66
/app/vendor/cakephp/cakephp/src/Datasource/EntityTrait.php:260
/app/vendor/cakephp/cakephp/src/ORM/Marshaller.php:222
/app/vendor/cakephp/cakephp/src/ORM/Table.php:2742
/app/vendor/vierge-noire/cakephp-fixture-factories/src/Factory/DataCompiler.php:231
/app/vendor/vierge-noire/cakephp-fixture-factories/src/Factory/DataCompiler.php:186
/app/vendor/vierge-noire/cakephp-fixture-factories/src/Factory/DataCompiler.php:170
/app/vendor/vierge-noire/cakephp-fixture-factories/src/Factory/BaseFactory.php:269
/app/vendor/vierge-noire/cakephp-fixture-factories/src/Factory/BaseFactory.php:224
/app/tests/TestCase/Controller/BankingProposalControllerTest.php:25
```

To fix it, I should load the `users` config file in my `tests/bootstrap.php` :

```
Configure::load('users');
```

I dont understand why I'm forced to do that way. I believed that since Cake 3.7 or nearly you should load plugins in `Application.php` and it will load the plugin configuration in bootstrap files ? 

Can someone explain it to me because I don't seem to understand 🤯

**Comment by pabloelcolombiano** [comment, marked as answer]
Hi @Petrucheqa,

thanks for the input. The answer lies in the stack trace :-)
`Application.php`'s bootstrap will indeed load the configuration. But you probably call the factory before the Application gets loaded.
In your factory's default template, you probably have a default password value. The setter in `Entity/User.php:66` will thus be triggered.
Therefor, the config is not loaded yet by the time you call the factory.


Since this config is vital for the UserFactory, I would load the config, if not loaded, in the `initialize()` method of the factory.

If this is not clear, you can paste your test, and I'll pinpoint the explanation above.
